### PR TITLE
add Seq2seq doc and more python seq2seq apis

### PR DESCRIPTION
**Pull request by dding3**
1. Added seq2seq document
2. Add python seq2seq.compile
3. Add python seq2seq.fit
4. Add python seq2seq.infer

### CODE REVIEW

File: docs/docs/APIGuide/Models/seq2seq.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
+Analytics Zoo provides Seq2seq model which is a general-purpose encoder-decoder framework that can be used for Chatbot, Machine Translation and more.
+The model could be fed into NNFrames or BigDL Optimizer directly for training.
+
+---
+## **Build a Seq2seq Model**
+Before build Seq2seq Model, you need build `Encoder`, `Decoder`. And `Bridge` if you want to do some transformation before pass encoder states to decoder.
```
Comment (hkvision) [inline docs/docs/APIGuide/Models/seq2seq.md]: pass -> passing

File: docs/docs/APIGuide/Models/seq2seq.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+### **Build an Encoder**
+Currently we only support `RNNEncoder` which enables put RNN layers into encoder.
```
Comment (hkvision) [inline docs/docs/APIGuide/Models/seq2seq.md]: enable put -> enables you to put?

File: docs/docs/APIGuide/Models/seq2seq.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+You can call the following API in Scala and Python respectively to create a `RNNEncoder`.
+
+**Scala**
+```scala
+val encoder = RNNEncoder(rnnType, numLayer, hiddenSize, embedding)
+```
+
+* `rnnType` style of recurrent unit, one of [SimpleRNN, LSTM, GRU]
+* `numLayers` number of layers used in encoder
```
Comment (hkvision) [inline docs/docs/APIGuide/Models/seq2seq.md]: numLayer or numLayers? The doc is inconsistent with the code

File: docs/docs/APIGuide/Models/seq2seq.md [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+## **Build a Seq2seq Model**
+Before build Seq2seq Model, you need build `Encoder`, `Decoder`. And `Bridge` if you want to do some transformation before pass encoder states to decoder.
+
+### **Build an Encoder**
+Currently we only support `RNNEncoder` which enables put RNN layers into encoder.
+You can call the following API in Scala and Python respectively to create a `RNNEncoder`.
+
+**Scala**
+```scala
+val encoder = RNNEncoder(rnnType, numLayer, hiddenSize, embedding)
```
Comment (hkvision) [inline docs/docs/APIGuide/Models/seq2seq.md]: any default value?

File: docs/docs/APIGuide/Models/seq2seq.md [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+```scala
+val encoder = RNNEncoder(rnns, embedding, inputShape)
+```
+
+* `rnns` rnn layers used for encoder, support stacked rnn layers
+* `embedding` embedding layer in encoder, `null` is supported
+
+**Python**
+```python
+encoder = RNNEncoder.initialize(rnn_type, nlayers, hidden_size, embedding):
```
Comment (hkvision) [inline docs/docs/APIGuide/Models/seq2seq.md]: remove ":"?

File: docs/docs/APIGuide/Models/seq2seq.md [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+
+* `rnn_type` style of recurrent unit, one of [SimpleRNN, LSTM, GRU]
+* `nlayers` number of layers used in encoder
+* `hidden_size` hidden size of encoder
+* `embedding` embedding layer in encoder, `None` is supported
+
+Or
+
+```python
+encoder = RNNEncoder(rnns, embedding)
```
Comment (hkvision) [inline docs/docs/APIGuide/Models/seq2seq.md]: wondering why python doesn't have input_shape but scala has?

File: docs/docs/ProgrammingGuide/seq2seq.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+* `rnn_type`: currently support "simplernn | lstm | gru"
+* `nlayers`: number of layers
+* `hidden_size`: hidden size
+* `bridge_type`: currently only support "dense | densenonlinear"
+* `input_shape`: shape of encoder input
+* `output_shape`: shape of decoder input
+
+---
+## **Train a Seq2seq model**
+After building the model, we can use BigDL Optimizer to train it (with validation) using RDD of [Sample](https://bigdl-project.github.io/master/#APIGuide/Data/#sample).
```
Comment (hkvision) [inline docs/docs/ProgrammingGuide/seq2seq.md]: does it have compile fit API?

File: docs/docs/ProgrammingGuide/seq2seq.md [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
+**Scala**
+```scala
+import com.intel.analytics.bigdl.optim._
+import com.intel.analytics.bigdl.nn.{TimeDistributedMaskCriterion, InternalClassNLLCriterion}
+
+val optimizer = Optimizer(
+model,
+trainSet,
```
Comment (hkvision) [inline docs/docs/ProgrammingGuide/seq2seq.md]: shall we elaborate a bit more about what trainSet is expected to be? i.e. what's the feature/label in the sample

File: docs/docs/ProgrammingGuide/seq2seq.md [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+TimeDistributedMaskCriterion(
+  InternalClassNLLCriterion(paddingValue = padId),
```
Comment (hkvision) [inline docs/docs/ProgrammingGuide/seq2seq.md]: shall we expose something Internal here?

File: docs/docs/APIGuide/Models/seq2seq.md (hunk 4 above)
Comment (dding3) [reply docs/docs/APIGuide/Models/seq2seq.md]: Added default value in the following parameter explanation.

File: docs/docs/ProgrammingGuide/seq2seq.md (hunk 7 above)
Comment (dding3) [reply docs/docs/ProgrammingGuide/seq2seq.md]: It's following Text_classification document :)

Comment (hkvision) [reply docs/docs/ProgrammingGuide/seq2seq.md]: lol... I am updating text classification using compile and fit on TextSet :)

Comment (dding3) [reply docs/docs/ProgrammingGuide/seq2seq.md]: Added compile fit api

File: docs/docs/ProgrammingGuide/seq2seq.md (hunk 8 above)
Comment (dding3) [reply docs/docs/ProgrammingGuide/seq2seq.md]: Added

**Comment by dding3** [comment]
jekins passed. http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/1754/

File: docs/docs/ProgrammingGuide/seq2seq.md [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+model.compile(optimizer, loss, metrics)
+
+model.fit(x, batch_size=32, nb_epoch=10, validation_data=None)
+```
+
+---
+## **Do prediction**
+
+**Scala**
+```scala
```
[the remark this answers is not in the document]
Comment (dding3) [reply docs/docs/ProgrammingGuide/seq2seq.md]: Added

**Comment by dding3** [comment]
Thanks for your review. Jekins passed. http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/1756/
